src/main/java/com/imsweb/validation/ValidationStagingUtils.java

```diff
@@ -0,0 +1,131 @@
+    
+    /**
+     * Private constructor, no instantiation.
+     */
+    private ValidationStagingUtils() {
+    }
+
+    public static String computeCsSchemaId(Staging stagingInstance, Map<String, String> input) {
+        if (stagingInstance == null || input == null)
+            return null;
+        
+        String site = input.get(INPUT_PROP_SITE);
+        String hist = input.get(INPUT_PROP_HIST);
+        String ssf25 = input.get(INPUT_PROP_SSF25);
+
+        SchemaLookup lkup = new SchemaLookup(site, hist);
+        lkup.setInput("ssf25", ssf25);
+        List<Schema> info = stagingInstance.lookupSchema(lkup);
+        if (info.size() == 1) {
+            Schema schema = stagingInstance.getSchema(info.get(0).getId());
+            return schema.getId();
+        }
+        return null;
+    }
+    
+    public static String computeTnmSchemaId(Staging stagingInstance, Map<String, String> input) {
+        if (stagingInstance == null || input == null)
+            return null;
+        
+        String site = input.get(INPUT_PROP_SITE);
+        String hist = input.get(INPUT_PROP_HIST);
+        String ssf25 = input.get(INPUT_PROP_SSF25);
+        String sex = input.get(INPUT_PROP_SEX);
+        
+        // get the TNM schema ID
+        SchemaLookup lkup = new SchemaLookup(site, hist);
+        lkup.setInput("ssf25", ssf25);
+        lkup.setInput("sex", sex);
+        List<Schema> info = stagingInstance.lookupSchema(lkup);
+        if (info.size() == 1) {
+            Schema schema = stagingInstance.getSchema(info.get(0).getId());
+            return schema.getId();
+        }
+        return null;
+    }
+
+    public static String computeEodSchemaId(Staging stagingInstance, Map<String, String> input) {
+        if (stagingInstance == null || input == null)
+            return null;
+            
+        String site = input.get(INPUT_PROP_SITE);
+        String hist = input.get(INPUT_PROP_HIST);
+        String disc1 = input.get(INPUT_PROP_DISC_1);
+        String disc2 = input.get(INPUT_PROP_DISC_2);
+        String sex = input.get(INPUT_PROP_SEX);
+        String behav = input.get(INPUT_PROP_BEHAV);
+        String dxYear = input.get(INPUT_PROP_DX_YEAR);
+
+        SchemaLookup lkup = new SchemaLookup(site, hist);
+        lkup.setInput("discriminator_1", disc1);
+        lkup.setInput("discriminator_2", disc2);
+        lkup.setInput("sex", sex);
+        lkup.setInput("behavior", behav);
+        lkup.setInput("year_dx", dxYear);
+        List<Schema> info = stagingInstance.lookupSchema(lkup);
+        if (info.size() == 1) {
+            Schema schema = stagingInstance.getSchema(info.get(0).getId());
+            return schema.getId();
+        }
+        return null;
+    }
+    
+    /**
+     * Returns the SSF25 value based on the sex value and some conditions
+     */
+    public static String getSsf25FromSex(String ssf25, String sex, String hist, String dxYear, String schemaId) {
```

Comment:
Is this only applicable to TNM? Or CS and TNM? Can you please make this clear in the description of the method. And then rename schemaId to tnmSchemaId, or csOrTnmSchemaId. Something to make the intend obvious.

Reply:
It should be tnmSchemaId. I'll rename it.